File: April-18/18-04-18/java/01-Exploring-ruins.java

import java.io.*;


class TestClass {
    public static void main(String args[] ) throws Exception {
        BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
        String str = input.readLine();
        int n = str.length();
        
        for(int i = 0; i < n; i++)
        {
            if(i+1 < n)
            {
                if(str.charAt(i) == '?' )
                {
                    if(str.charAt(i+1) == 'a' || ( i-1 >= 0 && str.charAt(i-1) == 'a'))
                        str = str.substring(0,i) + 'b' + str.substring(i+1);
                    else
                        str = str.substring(0,i) + 'a' + str.substring(i+1);
                }
            }
            if(i == n-1)
            {
                if(str.charAt(i) == '?')
                {
                    if(i-1 >= 0 && str.charAt(i-1) == 'a')
                        str = str.substring(0,i) + 'b' ;
                    else
                        str = str.substring(0,i) + 'a';
                        
                }
            }
        }
        System.out.println(str);
        
    }
}

File: April-18/18-04-18/python/01-Exploring-ruins.py

data = list(input())

n = len(data)

for i in range(0,n):
    if(i+1 < n):
        if(data[i] == '?'):
            if(data[i+1] == 'a' or data[i-1] == 'a'):
                data[i] = 'b'
            else:
                data[i] = 'a'
    elif(i == n -1):
        if(data[i] == '?'):
            if(data[i-1] == 'a'):
                data[i] = 'b'
            else:
                data[i] = 'a'
for i in data:
   print(i,end = '')
